44.md

```diff
@@ -0,0 +1,168 @@
+NIP-44
+======
+
+Encrypted Payloads (Versioned)
+------------------------------
+
+`optional` `author:paulmillr` `author:staab`
+
+The NIP introduces a new data format for keypair-based encryption. This NIP is versioned to allow multiple algorithm choices to exist simultaneously.
+
+The encrypted payload is encoded as base64. The first byte of the payload represents the version. The rest depends on the version.
+
+Currently defined encryption algorithms:
+
+- `0x00` - Reserved
+- `0x01` - XChaCha with same key `sha256(ecdh)` per conversation
+
+# Version 1
+
+Encoding:
+
+- `1` - 1 byte, version
+- `nonce` - 24 bytes
```

Comment:
it's important to note somewhere, perhaps here, that nonce should be fetched from CSPRNG (cryptographically secure random source). If we leave it up to users, they will use garbage and leak their messages through nonce reuse.